cmake/FlatccGenerateSources.cmake

```diff
@@ -90,15 +86,28 @@ cmake_minimum_required(VERSION 3.3)
 # TODO: not needed when cmake_minimum_required_version >= 3.4
 include(CMakeParseArguments)
 
-function(flatcc_generate_sources NAME)
+function(flatcc_generate_sources)
     # parse function arguments
     set(output_options SCHEMA COMMON COMMON_READER COMMON_BUILDER BUILDER READER VERIFIER JSON_PARSER JSON_PRINTER JSON)
     set(NO_VAL_ARGS ALL RECURSIVE ${output_options})
-    set(SINGLE_VAL_ARGS OUTPUT_DIR OUTFILE PREFIX TARGET)
+    set(SINGLE_VAL_ARGS OUTPUT_NAME OUTPUT_DIR OUTFILE PREFIX)
     set(MULTI_VAL_ARGS DEFINITION_FILES COMPILE_FLAGS PATHS)
 
     cmake_parse_arguments(FLATCC "${NO_VAL_ARGS}" "${SINGLE_VAL_ARGS}" "${MULTI_VAL_ARGS}" ${ARGN})
+    
+    # TODO: add following check when cmake_minimum_required_version >= 3.15
+    #if (FLATCC_KEYWORDS_MISSING_VALUES)
+    #    message(FATAL_ERROR "The following keywords in call to flatcc_generate_sources have no value: ${FLATCC_KEYWORDS_MISSING_VALUES}")
+    #endif()
 
+    if (FLATCC_UNPARSED_ARGUMENTS)
+        message(WARNING "The following unknown keywords in call to flatcc_generate_sources are ignored: ${FLATCC_UNPARSED_ARGUMENTS}")
+    endif()
```

Comment:
It's ok to have unparsed/missing arguments.
See e.g. the `output_options` variable above, or `OUTFILE`/`PREFIX`.

The only required arguments are:
- `NAME` for the name of the target
- `DEFINITION_FILES` for the files

The rest can be deduced.

Reply:
As far as I understand from the cmake documentation, FLATCC_UNPARSED_ARGUMENTS checks for arguments that are not in NO_VAL_ARGS, SINGLE_VAL_ARGS or MULTI_VAL_ARGS.
So for example when somebody calls 
`flatcc_generate_sources(somename DEFINITION_FILES mydef.fbs BLABLA)`
FLATCC_UNPARSED_ARGUMENTS will contain (only) 'BLABLA'. I tested that and it worked.

Reply:
Indeed, `_UNPARSED_ARGUMENTS` is useful!
Thanks for checking!